kafka/consumer/fetcher.py

```diff
@@ -664,19 +665,34 @@ def _create_fetch_requests(self):
                 log.debug("Adding fetch request for partition %s at offset %d",
                           partition, position)
 
-        if self.config['api_version'] >= (0, 10):
+        if self.config['api_version'] >= (0, 10, 1):
+            version = 3
+        elif self.config['api_version'] >= (0, 10):
             version = 2
         elif self.config['api_version'] == (0, 9):
             version = 1
         else:
             version = 0
         requests = {}
         for node_id, partition_data in six.iteritems(fetchable):
-            requests[node_id] = FetchRequest[version](
-                -1, # replica_id
-                self.config['fetch_max_wait_ms'],
-                self.config['fetch_min_bytes'],
-                partition_data.items())
+            if version < 3:
+                requests[node_id] = FetchRequest[version](
+                    -1,  # replica_id
+                    self.config['fetch_max_wait_ms'],
+                    self.config['fetch_min_bytes'],
+                    partition_data.items())
+            else:
+                # As of version == 3 partitions will be returned in order as
+                # they are requested, so to avoid starvation with
+                # `fetch_max_bytes` option we need this shuffle
+                partition_data = list(partition_data.items())
+                random.shuffle(partition_data)
```

Comment:
Weird, Github swallowed my earlier comment. 

+1 for explicitly shuffling to avoid py3.6 issues. 

However, since the protocol explicitly responds in the order requested, maybe this should be exposed to the user? Default to shuffling the order to avoid starvation, but allow the user to override that to always request a particular order... I can see someone using this as a sort of poor-man's priority-queue.  Not sure if this is a good idea, or if it'd require reworking a lot of of `kafka-python` internals... just a thought.